trailmap/parse_input.py

```diff
@@ -0,0 +1,125 @@
+import os
+import sys
+import subprocess
+import json
+import xml.etree.ElementTree as ET
+import cyclus
+
+
+def parse_input(input, commodity_dictionary):
+    '''Builds dicionary of facilities and their incommodies and outcommodities.
+
+    inputs:
+        - input: a Cyclus XML input file
+        - commodity_dictionary: a dictionary
+
+    outputs:
+        - facility_dictionary: a dictionary of each facility and its
+        incommodities and outcommodities. format:
+        {'facility' : (['incommodities'], ['outcommodities'])}
+    '''
+
+    tree = ET.parse(input)
+    root = tree.getroot()
+
+    input_archetypes = get_archetypes_in_input(root)
+    [facility_dict_in,
+     facility_dict_out] = get_facility_and_commod_names(root, input_archetypes,
+                                                        commodity_dictionary)
+
+    return facility_dict_in, facility_dict_out
+
+
+def get_facility_and_commod_names(root, input_archetypes,
+                                  commodity_dictionary):
+    '''reads Cyclus input file element tree and finds the facility names for
+    each facility specified in a <facility> tag. Uses the facility archetype
+    and module, the input and output commodities are retrieved.
+
+    inputs:
+        - root: root of a Cyclus input file tree
+        - archetypes_in_input: a dictionary
+        - commodity_dictionary: a dictionary
+    outputs:
+        - facility_dict: a dictionary of each facility and its
+        incommodities and outcommodities. format:
+        {'facility' : (['incommodities'], ['outcommodities'])}
+    '''
+    facility_dict_in = {}
+    facility_dict_out = {}
+
+    for facility in root.findall('./facility'):
+        facility_name = facility.find('name').text
+        facility_archetype = facility.find('config/').tag
+        facility_module = input_archetypes[facility_archetype]
+        commodities = commodity_dictionary[facility_module]
```

Comment:
Maybe you  should use `commodities[0]` and `commodities[1]` instead multiplying variables

Reply:
(you are not using `commodities`)
you use `in_commod_tags` and `out_commod_tags` instead

Reply:
Thank you, forgot to take that out